src/main/cpp/io/SwerveDriveIO.cpp

```diff
@@ -0,0 +1,176 @@
+
+#include "io/SwerveDriveIO.h"
+#include "Utils.h"
+
+void SwerveDriveIO::Periodic() {
+    // Process all the inputs and outputs to/from high level software.
+    VOKC_CALL(ProcessIO());
+}
+
+void SwerveDriveIO::SimulationPeriodic() {
+    // SimulationPeriodic
+}
+
+bool SwerveDriveIO::ProcessIO() {
+    OKC_CHECK(sw_interface_ != nullptr);
+    OKC_CHECK(hw_interface_ != nullptr);
+
+    // Set the software outputs
+    // If the swerve drive configuration needs to be updated, update it.
+    if (sw_interface_->update_config) {
+        OKC_CALL(UpdateDriveConfig(sw_interface_->drive_config));
+
+        // Lower the update flag
+        sw_interface_->update_config = false;
+    }
+
+    // If the encoders should be reset, reset them
+    if (sw_interface_->reset_drive_encoders) {
+        OKC_CALL(ResetDriveEncoders());
+
+        // Lower the encoder reset flag
+        sw_interface_->reset_drive_encoders = false;
+    }
+
+    if (sw_interface_->reset_steer_encoders) {
+        OKC_CALL(ResetSteerEncoders());
+
+        // Lower the encoder reset flag
+        sw_interface_->reset_steer_encoders = false;
+    }
+
+    // If the navX should be reset, reset it.
+    if (sw_interface_->reset_gyro) {
+        hw_interface_->ahrs->Reset();
+
+        // Lower the navX reset flag
+        sw_interface_->reset_gyro = false;
+    }
+
+    OKC_CHECK(hw_interface_->left_front_drive_motor != nullptr);
+    OKC_CHECK(hw_interface_->left_back_drive_motor != nullptr);
+    OKC_CHECK(hw_interface_->right_front_drive_motor != nullptr);
+    OKC_CHECK(hw_interface_->right_back_drive_motor != nullptr);
+
+    // Set the drive outputs.
+    // hw_interface_->left_front_drive_motor->Set(TeamOKC::Clamp(-this->sw_interface_->drive_config.max_output_drive, this->sw_interface_->drive_config.max_output_drive, &this->sw_interface_->left_front_drive_motor_output));
```

Comment:
Whatever is happening below here should definitely be in a separate function lol